src/openfe_analysis/utils/multistate.py

```diff
@@ -1,15 +1,61 @@
 import netCDF4 as nc
 import numpy as np
+from numpy.typing import NDArray
 from pathlib import Path
+import warnings
 from openff.units import unit
-from typing import Optional
+from typing import Optional, Tuple
 
-from . import __version__
+from openfe_analysis import __version__
+
+
+def _determine_position_indices(dataset: nc.Dataset) -> NDArray:
+    """
+    Determine which iteration indices hold positions.
+
+    Parameters
+    ----------
+    dataset : nc.Dataset
+      Dataset holding the MultiStateReporter generated NetCDF file.
+
+    Returns
+    -------
+    indices : NDArray[int]
+      An ordered array of iteration indices which hold positions.
+
+    Note
+    ----
+    This assumes that the indices are equally spaced by a given
+    value.
+    """
+    if hasattr(dataset, 'PositionInterval'):
+        indices = [
+            i for i in
+            range(0, dataset.dimensions['iteration'].size, dataset.PositionInterval)
+        ]
+    else:
+        wmsg = ('This is an older NetCDF file that does not yet contain '
```

Comment:
In the tests using the old file, could you wrap the universe creation with a pytest.warns that checks for this please?